.github/workflows/upgrade_test.yaml

```diff
@@ -0,0 +1,54 @@
+name: Upgrade functionality test
+
+on: 
+  push:
+    branches: [ upgrade-testing ]
+
+jobs:
+  
+  upgrade-functionality-test:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout action sources
+      uses: actions/checkout@v2.3.4
+
+    - name: Checkout fission sources
+      uses: actions/checkout@v2.3.4
+      with:
+        repository: fission/examples
+        path: examples
+
+    - name: Setup go
+      uses: actions/setup-go@v2.1.3
+      with:
+        go-version: '1.15.12'
+
+    - name: Setup Helm
+      uses: Azure/setup-helm@v1
+      with:
+        version: v3.3.4
+    
+    - name: Update helm
+      run: helm repo add helm https://charts.helm.sh/stable 
+
+    - name: Setup Skaffold 
+      run: |
+        curl -Lo skaffold https://storage.googleapis.com/skaffold/releases/latest/skaffold-linux-amd64
+        sudo install skaffold /usr/local/bin/
+        skaffold version
+
+    - name: Setup Kind Clutser
+      uses: engineerd/setup-kind@v0.5.0
+      with:
+        config: kind.yaml
+        
+    - name: Setup kubectl & fetch node information
+      run: |
+        kubectl cluster-info --context kind-kind
+        kind get kubeconfig --internal >$HOME/.kube/config
+        kubectl get nodes
+        sudo apt-get install -y apache2-utils
+
+    - name: Configure fission and run upgrade test
```

Comment:
Just wondering if it's possible to do each of these in an individual test:
* Install fission
* Create fission objects(functions, environments, http triggers, mqt, etc)
* Upgrade fission 
* Test the objects